File: app/src/main/java/com/jakebarnby/filemanager/sources/dropbox/DropboxFragment.java
package com.jakebarnby.filemanager.sources.dropbox;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jakebarnby.filemanager.R;
import com.jakebarnby.filemanager.sources.SourceFragment;
import com.jakebarnby.filemanager.glide.GlideApp;
import com.jakebarnby.filemanager.util.Constants;

/**
 * Created by Jake on 5/31/2017.
 */

public class DropboxFragment extends SourceFragment {

    private static final String TAG = "DROPBOX";

    /**
     * Return a new instance of this Fragment
     * @param sourceName The name of the source controlled by this fragment
     * @return A new instance of this fragment
     */
    public static SourceFragment newInstance(String sourceName) {
        SourceFragment fragment = new DropboxFragment();
        Bundle args = new Bundle();
        args.putString(Constants.FRAGMENT_TITLE, sourceName);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mSource = new DropboxSource(Constants.Sources.DROPBOX, this);
    }

    @Override
    public void onResume() {
        super.onResume();
        ((DropboxSource)getSource()).checkForAccessToken(getContext());
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/dropbox/DropboxFragment.kt
package com.jakebarnby.filemanager.sources.dropbox

import android.os.Bundle
import androidx.core.os.bundleOf
import com.jakebarnby.filemanager.models.SourceType
import com.jakebarnby.filemanager.ui.sources.SourceFragment
import com.jakebarnby.filemanager.util.Constants
import com.jakebarnby.filemanager.util.Constants.Sources

/**
 * Created by Jake on 5/31/2017.
 */
class DropboxFragment : SourceFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        presenter.setFileSource(DropboxSource(presenter))
    }

    override fun onResume() {
        super.onResume()
        (presenter.source as DropboxSource).checkForAccessToken(context!!)
    }

    companion object {
        private const val TAG = "DROPBOX"

        fun newInstance(sourceId: Int): SourceFragment =
            DropboxFragment().apply {
                arguments = bundleOf(
                    Constants.FRAGMENT_TITLE to SourceType.values()[sourceId]
                )
            }
    }
}